migrate.sh

```diff
@@ -0,0 +1,211 @@
+	# All plugins for ES 2.0.0 uses 3.0.0-SNAPSHOT version number
+	replaceLine "    <version>3.0.0-SNAPSHOT<\/version>" "" "plugins/$1/pom.xml"
+
+	# echo "### remove unused dev-tools and .git dirs and LICENSE.txt and CONTRIBUTING.md files"
+	rm -r plugins/$1/dev-tools
+	rm -rf plugins/$1/.git
+	rm plugins/$1/LICENSE.txt
+	rm plugins/$1/CONTRIBUTING.md
+
+	# echo "### commit changes"
+	git add .
+	git commit -m "add $1 module" > /dev/null
+}
+
+echo "# STEP 0 : prepare the job"
+
+# echo "## clean $DIR_TMP plugins dev-tools/target target"
+rm -rf $DIR_TMP 
+rm -rf plugins 
+rm -rf dev-tools/target
+rm -rf target
+
+# echo "## create git $GIT_BRANCH work branch"
+
+# It first clean the existing branch if any
+git branch -D $GIT_BRANCH > /dev/null
+
+# Create the new branch
+git branch $GIT_BRANCH > /dev/null
+git checkout $GIT_BRANCH > /dev/null 2>/dev/null
+
+echo "# STEP 1 : Core module"
+
+# create the tmp work dir
+# echo "## create $DIR_TMP temporary dir"
+mkdir $DIR_TMP
+
+# create the core module
+# echo "## create core module in $MODULE_CORE"
+rm -rf $MODULE_CORE 
+mkdir $MODULE_CORE
+# echo "## create $MODULE_CORE pom.xml"
+cp pom.xml $MODULE_CORE
+# echo "## modify $MODULE_CORE/pom.xml"
+# We move <parent></parent> block on top
+removeLines "<parent>" "<\/parent>" "$MODULE_CORE/pom.xml"
+insertLinesAfter "<\/modelVersion>" "    <parent>§        <groupId>org.elasticsearch<\/groupId>§        <artifactId>elasticsearch-parent<\/artifactId>§        <version>2.0.0-SNAPSHOT<\/version>§    <\/parent>§" "§" "$MODULE_CORE/pom.xml"
+# We clean useless data
+replaceLine "    <version>2.0.0-SNAPSHOT<\/version>" "" "$MODULE_CORE/pom.xml"
+removeLines "<inceptionYear>" "<\/scm>" "$MODULE_CORE/pom.xml"
+removeLines "<repositories>" "<\/repositories>" "$MODULE_CORE/pom.xml"
+
+# echo "## move src in $MODULE_CORE"
+git mv src/ $MODULE_CORE
+# echo "## move bin in $MODULE_CORE"
+git mv bin/ $MODULE_CORE
+# echo "## move config in $MODULE_CORE"
+git mv config/ $MODULE_CORE 
+# echo "## move lib in $MODULE_CORE"
+git mv lib/ $MODULE_CORE 
+# echo "## copy README.textile, LICENSE.txt and NOTICE.txt in $MODULE_CORE"
+cp README.textile $MODULE_CORE 
+cp LICENSE.txt $MODULE_CORE 
+cp NOTICE.txt $MODULE_CORE 
+# echo "## modify rest-api-spec location in $MODULE_CORE/pom.xml"
+replaceLine "                <directory>\${project.basedir}\/rest-api-spec<\/directory>" "                <directory>\${project.basedir}\/..\/rest-api-spec<\/directory>" "$MODULE_CORE/pom.xml"
+
+
+# echo "## commit changes"
+git add .
+git commit -m "create $MODULE_CORE module" > /dev/null
+
+echo "# STEP 2 : Parent pom.xml from $PARENT_GIT"
+
+# echo "## fetch parent project from $PARENT_GIT in $DIR_TMP"
+# If you want to run that locally, uncomment this line and comment one below
+# cp -r ../elasticsearch-parent $DIR_TMP
+git clone $PARENT_GIT $DIR_TMP/$PARENT_NAME > /dev/null 2>/dev/null
+
+cp $DIR_TMP/$PARENT_NAME/pom.xml .
+cp -r $DIR_TMP/$PARENT_NAME/dev-tools .
```

Comment:
Recommend changing all the `cp -r` to `cp -R`, Linux supports either, but the Mac OSX manpage for `cp` states:

```
Historic versions of the cp utility had a -r option.  This implementation supports
that option; however, its use is strongly discouraged, as it does not correctly copy
special files, symbolic links, or fifo's.
```